03_agent_deployment.sh

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env bash
+set -xe
+
+source logging.sh
+source common.sh
+source network.sh
+source utils.sh
+source validation.sh
+
+### Recommended configuration (compact scenario):
+### export IP_STACK=v4
+### export NETWORK_TYPE=OpenShiftSDN
+### export NUM_WORKERS=0
+### export MASTER_DISK=120
+### export MASTER_MEMORY=16384
+
+early_deploy_validation
+
+export FLEETING_ISO=${FLEETING_ISO:-/tmp/fleeting.iso}
+export FLEETING_PATH=${WORKING_DIR}/fleeting
+node0=${CLUSTER_NAME}_master_0
+node0_ip=$(nth_ip $EXTERNAL_SUBNET_V4 20)
+
+function generate_fleeting_manifests() {
+
+    FLEETING_MANIFESTS_PATH="${FLEETING_PATH}/manifests"
+
+    mkdir -p ${FLEETING_MANIFESTS_PATH}
+
+    cat > "${FLEETING_MANIFESTS_PATH}/agent-cluster-install.yaml" << EOF
+apiVersion: extensions.hive.openshift.io/v1beta1
+kind: AgentClusterInstall
+metadata:
+  name: test-agent-cluster-install
+  namespace: cluster0
+spec:
+  apiVIP: 192.168.111.10
+  ingressVIP: 192.168.111.11
+  clusterDeploymentRef:
+    name: compact-cluster
+  imageSetRef:
+    name: openshift-v4.10.0
+  networking:
+    clusterNetwork:
+    - cidr: 10.128.0.0/14
+      hostPrefix: 23
+    serviceNetwork:
+    - 172.30.0.0/16
+  provisionRequirements:
+    controlPlaneAgents: 3
```

Comment:
I guess this should be NUM_MASTERS - relatedly, do we expect the agent flow to allow SNO deployments?